include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -332,39 +427,134 @@ create_geometric_coarsening_sequence(Triangulation<dim, spacedim> const & fine_t
+      return size_shared;
+    }();
+
+
+    // Step 2: Create coarse grid on the first MPI process of each node
+    Triangulation<dim, spacedim> serial_tria;
+    replicate_distributed_triangulation_on_node(tria_copy,
+                                                mpi_comm,
+                                                n_processes_per_node,
+                                                serial_tria);
+
+    // Step 3: Continue as above but with the serial triangulation that gets
+    // distributed
+    unsigned int n_partitions = Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD);
+    for(unsigned int level = tria_copy.n_global_levels(); level > 0; --level)
+    {
+      // reduce the number of MPI ranks per coarsening step by at most a
+      // factor of 8, in order to avoid too much transfer out of a single MPI
+      // process during the MG level transfer.
+      n_partitions = std::min(
+        n_partitions,
+        std::max(n_partitions / 8,
+                 std::max<unsigned int>(1U, serial_tria.n_active_cells() / n_cells_per_process)));
 
       // extract relevant information from distributed triangulation
       auto const construction_data =
-        TriangulationDescription::Utilities::create_description_from_triangulation(
-          temp_tria, fine_triangulation->get_communicator());
+        TriangulationDescription::Utilities::create_description_from_triangulation_in_groups<dim,
+                                                                                             dim>(
+          [&](auto & tria) { tria.copy_triangulation(serial_tria); },
+          [&](auto & tria, auto const &, const auto) {
+#ifdef DEAL_II_WITH_METIS
+            GridTools::partition_triangulation(n_partitions, tria);
+#else
+            GridTools::partition_triangulation_zorder(n_partitions, tria);
+#endif
+          },
+          mpi_comm,
+          n_processes_per_node);
+
+      // create fully distributed triangulation
+      auto level_tria =
+        std::make_shared<parallel::fullydistributed::Triangulation<dim, spacedim>>(mpi_comm);
 
-      // actually create triangulation
-      new_tria->create_triangulation(construction_data);
+      for(auto const i : fine_triangulation->get_manifold_ids())
+        if(i != numbers::flat_manifold_id)
+          level_tria->set_manifold(i, fine_triangulation->get_manifold(i));
+      level_tria->create_triangulation(construction_data);
 
       // save mesh
-      coarse_grid_triangulations[l - 1] = new_tria;
+      coarse_grid_triangulations[level - 1] = level_tria;
+
+      // coarsen mesh
+      if(Utilities::MPI::this_mpi_process(mpi_comm) % n_processes_per_node == 0 && level > 1)
+        serial_tria.coarsen_global();
     }
   }
 
+  for(unsigned int i = 0; i < coarse_grid_triangulations.size(); ++i)
+    AssertThrow(i + 1 == coarse_grid_triangulations[i]->n_global_levels(),
+                ExcDimensionMismatch(i + 1, coarse_grid_triangulations[i]->n_global_levels()));
```

Comment:
I am not sure, but wouldn't a written message be more helpful?

Reply:
Done in e6b6fc41.